HARK/ConsumptionSaving/ConsIndShockModel.py

```diff
@@ -2392,12 +2348,76 @@ def checkConditions(self,verbose=False,public_call=True):
         else:
             self.FVAC = False
             print('The given type violates the Finite Value of Autarky Condition with the supplied parameter values. The FVAF is %2.4f' %(FVAF), end = " ")
-            violated = True
+            self.violated = True
             if public_call or verbose:
                 print('Therefore, a nondegenerate solution is not available (see '+url+'/#Conditions-Under-Which-the-Problem-Defines-a-Contraction-Mapping')
             print()
 
-        if verbose and violated:
+
+
+
+    def checkConditions(self,verbose=False,public_call=True):
+        '''
+        This method checks whether the instance's type satisfies the Absolute Impatience Condition (AIC), Weak Return
+        Impatience Condition (WRIC), Finite Human Wealth Condition (FHWC) and Finite Value of
+        Autarky Condition (FVAC).  When combinations of these conditions are satisfied, the 
+        solution to the problem exhibits different characteristics.  (For an exposition of the
+        conditions, see http://econ.jhu.edu/people/ccarroll/papers/BufferStockTheory/)
+
+        Parameters
+        ----------
+        verbose : boolean
+            Specifies different levels of verbosity of feedback. When False, it only reports whether the
+            instance's type fails to satisfy a particular condition. When True, it reports all results, i.e.
+            the factor values for all conditions.
+
+        Returns
+        -------
+        None
+        '''
+        violated = False # PerfForesightConsumerType.checkConditions(self, verbose=False, verbose_reference=False)
+
+        if self.cycles!=0 or self.T_cycle > 1:
+            return
+
+        # For theory, see hyperlink targets to expressions in 
+        # url=http://econ.jhu.edu/people/ccarroll/papers/BufferStockTheory
+        # For example, the hyperlink to the relevant section of the paper
+        url='http://econ.jhu.edu/people/ccarroll/papers/BufferStockTheory'
```

Comment:
ah wait I see this here, other functions like `checkGICInd` also use the var `url` can you make it `self.url` so that references can be passed?